stdlib/2and3/builtins.pyi

```diff
@@ -945,6 +945,8 @@ class dict(MutableMapping[_KT, _VT], Generic[_KT, _VT]):
     def __init__(self, map: Mapping[_KT, _VT], **kwargs: _VT) -> None: ...
     @overload
     def __init__(self, iterable: Iterable[Tuple[_KT, _VT]], **kwargs: _VT) -> None: ...
+    @overload
+    def __init__(self, **kwargs: Any) -> None: ...
```

Comment:
Will this overload ever match? I'd expect the first overload to always match instead:

```python
    @overload
    def __init__(self, **kwargs: _VT) -> None: ...
```

(Unfortunately, at least mypy is a bit overeager when determining base classes and will just infer `_VT` to be `object` when using `dict(x=3, y="foo")`.)

Reply:
That sounds to me like a bug in mypy. PEP 484 is pretty clear that when a TypeVar is matched, it should be matched consistently. So if _VT matches against an int value, then it needs to consistently match against an int.

Reply:
Could you give an example in which the new overload would be matched, not the first one?

Reply:
@erictraut 
> PEP 484 is pretty clear that when a TypeVar is matched, it should be matched consistently.

If you post a quote that makes you think so, we will fix it. Also see first example at https://www.python.org/dev/peps/pep-0483/#type-variables

> That sounds to me like a bug in mypy.

This is how inference for type variables work, they are not "matched", they are solved for. For example, if you have:
```python
def pack(one: T, other: T) -> List[T]:
    ...
```
then `pack(Cat(), Dog())` can be a `List[Animal]`.

Reply:
OK, thanks for the explanation.